### PR TITLE
Run build via `babel`

### PR DESCRIPTION
**Pull request by ogonkov**
- Replace custom UMD w/ `babel-plugin-transform-es2015-modules-umd`
- Replace `uglifyjs` w/ `babel-preset-minify`
- Move source file to `src` dir
- Add `module` field for use with bundlers ala `webpack`
- Generated files removed from repository (still be available in npm)

### CODE REVIEW

**Comment by ogonkov** [comment]
Fix #229 #299 #305 #362 

I believe using `babel` is much cleaner solution

**Comment by ogonkov** [comment]
I've revert export to the named one, since it's more [future proof](https://blog.neufund.org/why-we-have-banned-default-exports-and-you-should-do-the-same-d51fdc2cf2ad) for possible refactoring

**Comment by ogonkov** [comment]
@eligrey something wrong with my PR?

**Comment by eligrey** [comment]
@ogonkov It looks good to me, there's just a couple small points that need addressed first.

**Comment by ogonkov** [comment]
What points?

File: src/FileSaver.js [hunk 1]
```diff
@@ -13,7 +13,7 @@
 
 /*! @source http://purl.eligrey.com/github/FileSaver.js/blob/master/FileSaver.js */
 
-var saveAs = saveAs || (function(view) {
+export var saveAs = (function(view) {
```
[the remark this answers is not in the document]
Comment (ogonkov) [reply src/FileSaver.js]: This would be inside closure, so it seems doesn't required anymore

[the remark this answers is not in the document]
Comment (eligrey) [reply src/FileSaver.js]: If `saveAs` is already defined globally, FileSaver.js should not try to make its own. Please reinsert ` saveAs ||`.

[the remark this answers is not in the document]
Comment (ogonkov) [reply src/FileSaver.js]: With `babel-plugin-transform-es2015-modules-umd` it exports `FileSaver` to `window`, to have consistent exports signature with CommonJS modules.

**Comment by ogonkov** [comment]
Should i update `bower.json` and readme?

File: src/FileSaver.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (eligrey) [reply src/FileSaver.js]: I don't want to export anything if saveAs is already defined. Just so you understand if you want this merged you will need to include the `saveAs ||` part of the statement.

[the remark this answers is not in the document]
Comment (ogonkov) [reply src/FileSaver.js]: It's probably misunderstanding, saveAs is not being exported to `window`, it is inside es6 module. We define `saveAs` inside module scope, then export it from module, so it's never make its way to window, and it doesn't override `window.saveAs`.

[the remark this answers is not in the document]
Comment (eligrey) [reply src/FileSaver.js]: I'm saying that I want it to inherit from window.saveAs if it's already defined.

[the remark this answers is not in the document]
Comment (eligrey) [reply src/FileSaver.js]: `export var saveAs = saveAs || ...` will accomplish that.

File: src/FileSaver.js [hunk 2]
```diff
@@ -13,7 +13,7 @@
 
 /*! @source http://purl.eligrey.com/github/FileSaver.js/blob/master/FileSaver.js */
 
-var saveAs = saveAs || (function(view) {
+export var saveAs = 'saveAs' in window ? window.saveAs : (function(view) {
```
Comment (ogonkov) [inline src/FileSaver.js]: I have add explicit check for `window.saveAs`

**Comment by ogonkov** [comment]
Anything more i can improve here @eligrey ?

**Comment by eligrey** [comment]
Yes, you can do the change I required. This won't be merged until the export line starts with `export var saveAs = saveAs || ...`

**Comment by ogonkov** [comment]
@eligrey done. Any other changes i should make?

File: src/FileSaver.js [hunk 3]
```diff
@@ -13,7 +13,7 @@
 
 /*! @source http://purl.eligrey.com/github/FileSaver.js/blob/master/FileSaver.js */
 
-var saveAs = saveAs || (function(view) {
+export var saveAs = saveAs || (function(view) {
```
Comment (graingert) [inline src/FileSaver.js]: can you do this as export default?

Comment (eligrey) [reply src/FileSaver.js]: Oh, I wasn't even aware of export default. Yeah, this should use default, sorry for jumping the gun on merging.

File: src/FileSaver.js [hunk 4]
```diff
@@ -175,11 +175,3 @@ var saveAs = saveAs || (function(view) {
 	|| typeof window !== "undefined" && window
 	|| this
```
Comment (graingert) [inline src/FileSaver.js]: `this` is undefined in ESM

Comment (eligrey) [reply src/FileSaver.js]: I know that. It's also the global variable in some contexts that this script could run, although probably not through babel.

Comment (graingert) [reply src/FileSaver.js]: Yes Babel will mark this as undefined.

**Comment by harshpatel991** [comment]
@eligrey When will these changes go into a release? My project depends on a fix made here (specifically #383 )

Looks like the latest change in the repo is a couple months old.

**Pull request merged**